### PR TITLE
Snapshot size calculation fix

### PR DESCRIPTION
**Pull request by jumaffre**
Noticed this when plugging the new snapshot mechanism into the rest of CCF. The size of the pre-insertion serialised K,V pair returned by `put_mut()` was actually the size of the post-insertion serialised K,V. This worked so far in our unit tests because we use same keys and values but didn't in the case of the `ccf.member_acks` table which changes in size as members ack.

### CODE REVIEW

File: src/ds/champ_map.h [hunk 1]
```diff
@@ -498,7 +500,7 @@ namespace champ
         "size:{}, map->size:{} ==> count:{}, vect:{}",
```
Comment (jumaffre) [inline src/ds/champ_map.h:500]: This assert has certainly saved hours of debugging!

**Comment by eddyashton** [comment]
> This worked so far in our unit tests because we use same keys and values

Can we add a unit test that uses different key/value types to stress this?

**Comment by jumaffre** [comment]
> > This worked so far in our unit tests because we use same keys and values
> 
> Can we add a unit test that uses different key/value types to stress this?

@eddyashton Good point. I have added a unit test for this.

File: src/ds/test/map_test.cpp [hunk 2]
```diff
@@ -140,12 +140,9 @@ TEST_CASE("persistent map operations")
 static const champ::Map<K, V, H> gen_map(size_t size)
 {
   champ::Map<K, V, H> map;
-  for (uint32_t j = 0; j < 2; ++j)
+  for (uint64_t i = 0; i < size; ++i)
```
Comment (achamayou) [inline src/ds/test/map_test.cpp:143]: size_t

**Pull request merged**